src/components/BusinessDonationHistory/BusinessDonationHistory.jsx

```diff
@@ -2,26 +2,67 @@
 
 import { useEffect, useState } from 'react';
 import { useBackend } from '../../contexts/BackendContext';
-import { Table, Thead, Tbody, Tr, Td, Button, Th } from '@chakra-ui/react';
-import ViewDonationHistory from './ViewDonationHistory/ViewDonationHistory';
+import { useAuth } from '../../contexts/AuthContext';
+import {
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Td,
+  Button,
+  Th,
+  Box,
+  IconButton,
+  InputGroup,
+  Input,
+  Heading,
+  Flex,
+  Spacer,
+  TableContainer,
+} from '@chakra-ui/react';
+import { useNavigate } from 'react-router-dom';
+import { ChevronRightIcon, ChevronLeftIcon, ArrowForwardIcon } from '@chakra-ui/icons';
+import classes from './BusinessDonationHistory.module.css';
 
 const BusinessDonationHistory = () => {
   const { backend } = useBackend();
+  const { currentUser } = useAuth();
   const [data, setData] = useState([]);
   const [selectedDonationId, setSelectedDonationId] = useState(null);
-  const TABLE_HEADERS = ['Submitted By', 'Food Bank', 'Date', 'Action'];
+  const navigate = useNavigate();
+
+  // for pagination
+  const [currentTotalDonationNum, setCurrentTotalDonationNum] = useState(0);
+  const [currentPageNum, setCurrentPageNum] = useState(1);
+  const [pageLimit, setPageLimit] = useState(1);
+
+  // for search
+  const [searchInput, setSearchInput] = useState('');
+
+  // eslint-disable-next-line no-unused-vars
+  const [paginationNumber, setPaginationNumber] = useState(10);
 
   useEffect(() => {
     const getData = async () => {
       try {
-        const response = await backend.get('/donation');
+        // Fetch business ID from backend
+        const businessIdResponse = await backend.get(`/businessUser/${currentUser.uid}`);
+        const response = await backend.get(
+          `/donation/filter/search/?businessId=${businessIdResponse.data[0].id}&searchTerm=${searchInput}&donationsLimit=${paginationNumber}&pageNum=${currentPageNum}`,
+        );
         setData(response.data);
+        const donationNumResponse = await backend.get(
+          `/donation/filter/searchCount/?businessId=${businessIdResponse.data[0].id}&searchTerm=${searchInput}`,
+        );
+        setCurrentTotalDonationNum(donationNumResponse.data[0]['count']);
+        setPageLimit(Math.ceil(donationNumResponse.data[0]['count'] / paginationNumber));
+
       } catch (error) {
         console.error('Error fetching data:', error);
       }
     };
     getData();
-  }, []);
+  }, [backend, currentPageNum, searchInput, paginationNumber]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'currentUser.uid'. Either include it or remove the dependency array.